File: yh_star/src/yh_star_pub.cpp
#include "ros/ros.h"
#include "yh_star/YhStarMsg.h"   //패키지 이름,만든 메세지 파일.


int main(int argc,char**argv)
{
    ros::init(argc,argv, "yh_star_pub");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<yh_star::YhStarMsg>("yh_star_topic",20);
    
   
    ros::Rate loop_rate(2); //1초에 2번 0.5초에 1번  Hz주기

    yh_star::YhStarMsg msg;
    int cnt = 1;


    while(ros::ok())
    {
        msg.stamp = ros::Time::now();
        msg.data = cnt;  
        pub.publish(msg);
        cnt++;
        loop_rate.sleep();
    }
     return 0;    

}

File: yh_star/scripts/yh_star_pub.py
#!/usr/bin/python3
#-*- coding: utf-8 -*- 

import rospy
from yh_star.msg import YhStarMsg

def star_publisher():
    rospy.init_node("yh_star_pub")
    pub = rospy.Publisher("yh_star_topic", YhStarMsg,queue_size=100)

    loop_rate = rospy.Rate(2)

    msg = YhStarMsg()


    while not rospy.is_shutdown():

        msg.stamp = rospy.Time.now()
        msg.data += 1
        pub.publish(msg)
        loop_rate.sleep()

if __name__ == "__main__":
    try:
        star_publisher()
    except rospy.ROSInterruptException:
            pass


        

